core/src/main/scala/kafka/log/Log.scala

```diff
@@ -792,6 +798,34 @@ class Log(@volatile var dir: File,
     deleteOldSegments(shouldDelete)
   }
 
+  private def updateAppendInfoForDuplicateEntry(appendInfo: LogAppendInfo) : LogAppendInfo = {
```

Comment:
It seems this method doesn't take the requested PID into account. It only checks sequence number.